code/modules/spells/spell_code.dm

```diff
@@ -116,7 +124,49 @@ var/list/spells = typesof(/spell) //needed for the badmin verb for now
 			cast(targets, user)
 		after_cast(targets) //generates the sparks, smoke, target messages etc.
 
+/spell/proc/channel_spell(mob/user = usr, skipcharge = 0, force_remove = 0)
+	if(!holder)
+		holder = user //just in case
+	if(!user.spell_channeling && !force_remove)
+		if(!cast_check(skipcharge, user))
+			return 0
+		user.spell_channeling = user.on_uattack.Add(src, "channeled_spell")
+		connected_button.name = "(Ready) [name]"
+		currently_channeled = 1
+		connected_button.add_channeling()
+	else
+		var/event/E = user.on_uattack
+		E.handlers.Remove(user.spell_channeling)
+		user.spell_channeling = null
+		charge_counter = charge_max
+		currently_channeled = 0
+		connected_button.name = name
+		connected_button.remove_channeling()
+	return 1
 
+/spell/proc/channeled_spell(var/list/args)
+	var/event/E = args["event"]
+	if(!currently_channeled)
+		E.handlers.Remove("\ref[src]:channeled_spell")
+		return
+
+	var/atom/A = args["atom"]
+
+	if(E.holder != holder)
+		E.handlers.Remove("\ref[src]:channeled_spell")
+		return
+	var/list/target = list(A)
+	holder:attack_delayer.delayNext(0)
```

Comment:
You forgot to say 'where did you copypaste this from' and the answer was lightning